back/app/models/basket.rb

```diff
@@ -46,6 +46,15 @@ def total_votes
     baskets_ideas.pluck(:votes).sum
   end
 
+  def destroy_or_keep!
+    if submitted_at.present? && participation_context_type == 'Phase' && participation_context.end_at <= Time.now
```

Comment:
The check that the phase finished should use the tenant's timezone. Perhaps there are some helper methods for this around?

I also introduced a `submitted?` predicate.